File: src/main/java/com/gmail/borlandlp/minigamesdtools/creator/ICreatorHub.java
package com.gmail.borlandlp.minigamesdtools.creator;

import java.util.List;

public interface ICreatorHub {
    void registerCreator(Creator factory) throws Exception;
    void registerRouteId2Creator(String itemID, String creator) throws Exception;
    Object create(String itemID, AbstractDataProvider dataProvider) throws Exception;
    void unregisterCreator(Creator creator);
    void unregisterCreator(String creatorID);
    void unregisterRouteId2Creator(String itemID);
    boolean containsCreator(String ID);
    boolean containsCreator(Creator creator);
    boolean containsRouteId2Creator(String itemId);
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/creator/ICreatorHub.kt
package com.gmail.borlandlp.minigamesdtools.creator

interface ICreatorHub {
    @Throws(Exception::class)
    fun registerCreator(factory: Creator)

    @Throws(Exception::class)
    fun registerRouteId2Creator(itemID: String, creator: String)

    @Throws(Exception::class)
    fun create(itemID: String, dataProvider: AbstractDataProvider): Any

    fun unregisterCreator(creator: Creator)
    fun unregisterCreator(creatorID: String)
    fun unregisterRouteId2Creator(itemID: String)
    fun containsCreator(ID: String): Boolean
    fun containsCreator(creator: Creator): Boolean
    fun containsRouteId2Creator(itemId: String): Boolean
}